core/rawdb/accessors_chain.go

```diff
@@ -64,6 +64,26 @@ func DeleteCanonicalHash(db ethdb.KeyValueWriter, number uint64) {
 	}
 }
 
+func IsTrustBlock(db ethdb.Reader, hash common.Hash) bool {
+	data, _ := db.Get(trustBlockHashKey(hash))
+	if len(data) == 0 {
```

Comment:
what about:
```
if len(data) != 1{
   return false
}
return data[0] == byteTrue 
```

We can save the allocation of slices each time, more memory friendly.